keras_contrib/applications/densenet.py

```diff
@@ -323,3 +467,146 @@ def __create_dense_net(nb_classes, img_input, include_top, depth=40, nb_dense_bl
+
+        final_nb_layer = nb_layers[-1]
+        nb_layers = nb_layers[:-1]
+
+    else:
+        final_nb_layer = nb_layers_per_block
+        nb_layers = [nb_layers_per_block] * nb_dense_block
+
+    if bottleneck:
+        nb_layers = [int(layer // 2) for layer in nb_layers]
+
+    # compute initial nb_filter if -1, else accept users initial nb_filter
+    if nb_filter <= 0:
+        nb_filter = 2 * growth_rate
+
+    # compute compression factor
+    compression = 1.0 - reduction
+
+    # Initial convolution
+    x = Convolution2D(48, 3, 3, init="he_uniform", border_mode="same", name="initial_conv2D", bias=False,
+                      W_regularizer=l2(weight_decay))(img_input)
+
+    skip_connection = x
+    skip_list = []
+
+    # Add dense blocks and transition down block
+    for block_idx in range(nb_dense_block):
+        x, nb_filter = __dense_block(x, nb_layers[block_idx], nb_filter, growth_rate, bottleneck=bottleneck,
+                                     dropout_rate=dropout_rate, weight_decay=weight_decay)
+
+        # Skip connection
+        x = merge([x, skip_connection], mode='concat', concat_axis=concat_axis)
+        skip_list.append(x)
+
+        # add transition_block
+        x = __transition_block(x, nb_filter, compression=compression, dropout_rate=dropout_rate,
+                               weight_decay=weight_decay)
+        nb_filter = int(nb_filter * compression)  # this is calculated inside transition_down_block
+
+        # Preserve transition for next skip connection after dense
+        skip_connection = x
+
+    # The last dense_block does not have a transition_down_block
+    x, nb_filter = __dense_block(x, final_nb_layer, nb_filter, growth_rate, bottleneck=bottleneck,
+                                 dropout_rate=dropout_rate, weight_decay=weight_decay)
+
+    if K.image_dim_ordering() == 'th':
+        out_shape = [batchsize, nb_filter, rows // 16, cols // 16]
+    else:
+        out_shape = [batchsize, rows // 16, cols // 16, nb_filter]
+
+    # Add dense blocks and transition up block
+    for block_idx in range(nb_dense_block):
+        if K.image_dim_ordering() != 'th':
+            out_shape[3] = nb_filter
+
+        x = __transition_up_block(x, nb_filters=nb_filter, type=upsampling_type, output_shape=out_shape)
+
+        if K.image_dim_ordering() == 'th':
+            out_shape[2] *= 2
+            out_shape[3] *= 2
+        else:
+            out_shape[1] *= 2
+            out_shape[2] *= 2
+
+        x = merge([x, skip_list.pop()], mode='concat', concat_axis=concat_axis)
+
+        x, nb_filter = __dense_block(x, nb_layers[-block_idx], nb_filter, growth_rate, bottleneck=bottleneck,
+                                     dropout_rate=dropout_rate, weight_decay=weight_decay)
+
+    x = Convolution2D(nb_classes, 1, 1, activation='linear', border_mode='same', W_regularizer=l2(weight_decay),
+                      bias=False)(x)
+
+    if K.image_dim_ordering() == 'th':
```

Comment:
Yes, that is correct if your problem domain is strictly segmentation for evaluating an FCN, but I'm going to incorporate additional layers because FCN is merely a component of my larger network. If you prefer I can make that a future pull request after this is merged.